force-app/main/default/classes/ApexBasicAssignment.cls

```diff
@@ -0,0 +1,39 @@
+public class ApexBasicAssignment {
+	public static List<Student__c> getStudentsList(String classId){
+        List<Student__c> studentList = [SELECT Name FROM Student__c WHERE Class__r.Name =: classId];
+        return studentList;
+    }
+    
+    public static Map<String,List<Student__c>> mapClassAndStudents(){
+        Map<String,List<Student__c>>classAndStudents = new Map<String,List<Student__c>>();
+        List<Class__c>  listClass = [SELECT Name FROM Class__c];
+        List<List<Student__c>> studentsId = new List<List<Student__c>>();
```

Comment:
Do we really need to maintain List<List<Student__c>> here?
We can simply do this:
```
for(Class__c cls : listClass)
{
            List<Student__c> studentsList = [SELECT Id ,Name FROM Student__c WHERE Class__r.Name =: cls.Name];
            classAndStudents.put(cls.Name, studentsList);          
        }
```